### PR TITLE
diff_fasm: speedup generation of vendor tools bitstreams

### PR DESCRIPTION
**Pull request by acomodi**
Signed-off-by: Alessandro Comodi <acomodi@antmicro.com>

### CODE REVIEW

**Comment by mithro** [comment]
@acomodi -- What type of impact does this have on the speed?

File: xc/common/utils/vivado_create_runme.py [hunk 1]
```diff
@@ -47,16 +47,47 @@ def create_runme(f_out, args):
...
+if {$conflict_nets != ""} {
+    puts "There are conflicting nets!"
+
+    foreach net $conflict_nets {
+        puts $net
+    }
+    error "Exiting!"
+}
+
+set unrouted_nets [get_nets -filter { ROUTE_STATUS == "UNDROUTED" }]
```
Comment (litghost) [inline xc/common/utils/vivado_create_runme.py:67]: Misspelled UNROUTED?

**Comment by litghost** [comment]
@acomodi Have you profiled diff fasm targets to verify the problem lies in the routing step?

**Comment by acomodi** [comment]
I have gathered runtimes for the baselitex test.

Current run time of diff_fasm:
- source tcl: 10m15s
- place: 20m43s
- route: 2m41s
- write bitstream: 2m33s
- total time: 36m12s

With this PR:
- source tcl: 10m13s
- place: 0m0s (no placement performed)
- route: 0m16s
- write bitstream: 2m25s
- total: 12m54s

**Comment by litghost** [comment]
> I have gathered runtimes for the baselitex test.
> 
> Current run time of diff_fasm:
> 
> * source tcl: 10m15s
> * place: 20m43s
> * route: 2m41s
> * write bitstream: 2m33s
> * total time: 36m12s
> 
> With this PR:
> 
> * source tcl: 10m13s
> * place: 0m0s (no placement performed)
> * route: 0m16s
> * write bitstream: 2m25s
> * total: 12m54s
> 
> We get a huge improvement by just avoiding the place step which is the most time consuming and is not actually needed.
> As discussed, the `source tcl` step is the next to improve.

What this data shows is that avoiding placement (or accelerating it) is the first key modification.  The changes to the route step are helpful, but I'd argue for leaving it alone until the place and source tcl time is down.

I'm wondering whether the route time drop is primilarily the focusing on certain nets, or disabling some of the DRC checks?

**Comment by litghost** [comment]
@acomodi Should this be rebased?  I'm seeing a lot more changes than just the diff_fasm changes?

**Comment by acomodi** [comment]
> @acomodi Should this be rebased? I'm seeing a lot more changes than just the diff_fasm changes?

Yeah I'll rebase now

File: xc/common/utils/vivado_create_runme.py [hunk 2]
```diff
@@ -65,9 +93,8 @@ def create_runme(f_out, args):
 
 write_checkpoint -force design_{name}.dcp
 write_bitstream -force design_{name}.bit
-save_project_as -force design_{name}.xpr
```
Comment (litghost) [inline xc/common/utils/vivado_create_runme.py:68]: Don't remove the XPR, that is needed for simulation in ModelSim.

Comment (acomodi) [reply xc/common/utils/vivado_create_runme.py:68]: Sure, done

**Comment by acomodi** [comment]
This lets XC7 vendor tools to take ~38 minutes less w.r.t. to master
For XC7 200 the time gained is ~20 minutes though.

Not the best improvement, but it is still worth more than nothing.

Next step would be to use the XDC constraints instead of sourcing the tcl

File: xc/common/utils/vivado_create_runme.py (hunk 1 above)
Comment (acomodi) [reply xc/common/utils/vivado_create_runme.py]: Done

**Pull request merged**